AArch64 builds fail when building Redis (GCC >= 12)

### Describe the bug

Building [`app-redis`](https://github.com/unikraft/app-redis) with GCC >= 12 for AArch64 results in an error. This was introduced by PR #1009.

Note this is a different issue than issue #1129, which doesn't happen on GCC >= 12.

### Steps to reproduce

Use the commands below to reproduce the issue:

```console
git clone https://github.com/unikraft/app-redis redis
cd redis/
./scripts/setup.sh
wget https://raw.githubusercontent.com/unikraft/app-testing/staging/scripts/generate.py -O scripts/generate.py
chmod a+x scripts/generate.py
./scripts/generate.py
./scripts/build/make-qemu-arm64-9pfs.sh
```

### Expected behavior

The Redis build should work on AArch64.

### Which architectures were you using or does this bug affect?

arm64

### Which platforms were you using or does this bug affect?

kvm

### Relevant log output

```shell
LD      redis_qemu-arm64.dbg
/usr/lib/gcc-cross/aarch64-linux-gnu/12/../../../../aarch64-linux-gnu/bin/ld: warning: -z relro ignored
/usr/lib/gcc-cross/aarch64-linux-gnu/12/../../../../aarch64-linux-gnu/bin/ld: /home/unikraft/maintainer-tools/workdir/apps/redis/workdir/build/libkvmplat.o: in function `arm64_syscall_adapter':
/home/unikraft/maintainer-tools/workdir/unikraft/plat/common/arm/traps_arm64.c:233: undefined reference to `ukplat_syscall_handler'
/home/unikraft/maintainer-tools/workdir/unikraft/plat/common/arm/traps_arm64.c:233:(.text+0xf5c): relocation truncated to fit: R_AARCH64_CALL26 against undefined symbol `ukplat_syscall_handler'
collect2: error: ld returned 1 exit status
make[3]: *** [/home/unikraft/maintainer-tools/workdir/unikraft/plat/kvm/Linker.uk:50: /home/unikraft/maintainer-tools/workdir/apps/redis/workdir/build/redis_qemu-arm64.dbg] Error 1
```

--- a/plat/common/arm/traps_arm64.c
+++ b/plat/common/arm/traps_arm64.c
@@ -222,7 +222,7 @@ void trap_el1_irq(struct __regs *regs)
 	gic->ops.handle_irq(regs);
 }
 
-#ifdef CONFIG_LIBSYSCALL_SHIM
+#ifdef CONFIG_LIBSYSCALL_SHIM_HANDLER
 
 extern void ukplat_syscall_handler(struct __regs *r);
 
@@ -236,4 +236,4 @@ static int arm64_syscall_adapter(void *data)
 
 UK_EVENT_HANDLER(UKARCH_TRAP_SYSCALL, arm64_syscall_adapter);
 
-#endif /* CONFIG_LIBSYSCALL_SHIM */
+#endif /* CONFIG_LIBSYSCALL_SHIM_HANDLER */

